src/apps/chifra/pkg/types/types_transaction.go

```diff
@@ -0,0 +1,187 @@
+	MaxPriorityFeePerGas Gas             `json:"maxPriorityFeePerGas"`
+	Input                string          `json:"input"`
+	IsError              bool            `json:"isError"`
+	HasToken             bool            `json:"hasToken"`
+	Cachebits            uint8           `json:"cachebits"`
+	Reserved2            uint8           `json:"reserved2"`
+	Receipt              *SimpleReceipt  `json:"receipt"`
+	Traces               []SimpleTrace   `json:"traces"`
+	ArticulatedTx        *SimpleFunction `json:"articulatedTx"`
+	raw                  *RawTransaction
+}
+
+func (s *SimpleTransaction) Raw() *RawTransaction {
+	return s.raw
+}
+
+func (s *SimpleTransaction) SetRaw(raw *RawTransaction) {
+	s.raw = raw
+}
+
+func (s *SimpleTransaction) Model(showHidden bool, format string, extraOptions map[string]any) Model {
+	// EXISTING_CODE
+	to := hexutil.Encode(s.To.Bytes())
+	if to == "0x0000000000000000000000000000000000000000" {
+		to = "0x0" // weird special case to preserve what RPC does
+	}
+
+	// TODO: these date-related values could be done when RPC is queried and cached
+	date := gostradamus.FromUnixTimestamp(int64(s.Timestamp))
+
+	// TODO: Imporant note. The `finalized` field creates a dependacy on the transaction model that
+	// TODO: it is in a block (or at least we know the blockNumber. Also, who's to say what `finalized`
+	// TODO: means? Also, `finalized` has a meaning in post-merge code. See #2667
+	// EXISTING_CODE
+
+	model := map[string]interface{}{
+		"blockNumber":      s.BlockNumber,
+		"transactionIndex": s.TransactionIndex,
+		"timestamp":        s.Timestamp,
+		"from":             s.From,
+		"to":               to,
+		"gasUsed":          s.GasUsed,
+		"hash":             s.Hash,
+		"isError":          s.IsError,
+		"finalized":        extraOptions["finalized"],
+	}
+
+	order := []string{
+		"blockNumber",
+		"transactionIndex",
+		"date",
+		"timestamp",
+		"from",
+		"to",
+		"ether",
+		"ethGasPrice",
+		"gasUsed",
+		"hash",
+		"isError",
+		"encoding",
+		"compressedTx",
+	}
+
+	// EXISTING_CODE
+	if format == "json" {
+		model["blockHash"] = s.BlockHash
+		model["nonce"] = s.Nonce
+		model["value"] = s.Value
+		model["gas"] = s.Gas
+
+		// TODO: this value could be created when RPC is queried and cached
+		model["ether"] = weiToEther(&s.Value)
+		model["gasPrice"] = s.GasPrice
+		model["maxFeePerGas"] = s.MaxPriorityFeePerGas
+		model["maxPriorityFeePerGas"] = s.MaxPriorityFeePerGas
+		model["input"] = s.Input
+		model["hasToken"] = s.HasToken
+		model["receipt"] = s.Receipt
+		model["value"] = s.Value.String()
+		model["receipt"] = nil
+
+		// TODO: these date-related values could be done when RPC is queried and cached
+		model["date"] = date.Format("2006-01-02 15:04:05") + " UTC"
+		model["datesh"] = date.Format("2006-01-02")
+		model["time"] = date.Format("15:04:05") + " UTC"
+
+		if s.Receipt != nil {
+			// TODO: this should be a utility (and used above as well). May be available in go-ethereum. We should check.
```

Comment:
I think it's something along the lines of
```
if s.Receipt.ContractAddress == common.HexToAddress("0x0") {
```

Reply:
I feel like different node software might represent the zero address differently, so we should be permissions.

For example, I've seen `null`, `nil, `0x`, `0x0`, and `0x0000000000000000000000000000000000000000` over the years. Even Erigon itself has (mistakenly) popped back and forth in the past for some fields.

It's probably less of a problem now, but who knows. In the C++ code, there's a function called `isZeroAddress`.